### PR TITLE
js-oop(frogger)

### PR DESCRIPTION
**Pull request by alexdefender**
_Complete the form below and remove this line._

### task name

[Demo](https://alexdefender.github.io/frogger-arcade-game/) |
[Code base](https://github.com/alexdefender/frogger-arcade-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/alexdefender/frogger-arcade-game/app.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function (dt) {
+    this.x += this.speed * dt;
+    if (this.x > 500) {
+        this.x = -100;
+        this.speed = Math.random() * (500 - 50) + 50;
```
Comment (OleksiyRudenko) [inline submissions/alexdefender/frogger-arcade-game/app.js]: Mysterious numbers. Please, define and use constants.
Here and elsewhere across your code.

   * [ ] all numbers like block dimensions, initial locations are defined as constants
   * [ ] there are core constants and derived constants
         (e.g. `const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;`)
   * [ ] arrays of constants are also constants 
         (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)
   * [ ] const objects help organizing and structure const data even better
         (e.g. `const PLAYER_CONF = { initialPosition: {x: 1, y: 5}, sprite: '...', ...etc... };`

File: submissions/alexdefender/frogger-arcade-game/app.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    if ((player.y + 5 === this.y || player.y + 13 === this.y || player.y + 20 === this.y)
+        && player.x < this.x + 50 && player.x > this.x - 50) {
+        player.goStartPosition();
```
Comment (OleksiyRudenko) [inline submissions/alexdefender/frogger-arcade-game/app.js]: Classes should not refer to any global variables, like global variable `player`, which is an instance of `Player` class (referring to global constants and globals provided by the gaming platform like `Resources` is OK);

Hint: pass `Player` instance as an argument to every enemy

Comment (OleksiyRudenko) [inline submissions/alexdefender/frogger-arcade-game/app.js]: Separation of Concerns principle should be followed (e.g. `update` method does only rendering and doesn't **contain** any inline code to check e.g. collisions)

**Comment by alexdefender** [comment]
@OleksiyRudenko please review

File: submissions/alexdefender/frogger-arcade-game/app.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    this.collision();
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.collision = function () {
```
Comment (OleksiyRudenko) [inline submissions/alexdefender/frogger-arcade-game/app.js]: By convention function names start with verbs as  functions denote actions.

Comment (alexdefender) [reply submissions/alexdefender/frogger-arcade-game/app.js]: Thx.

**Pull request merged**